### ISSUE TITLE
VideoTrack migration 2.6.0

### ISSUE THREAD

**Issue by Max0u** [twilio/twilio-video-ios#33]
### Description

We just migrate Twilio to the latest version 2.6.0. We use VideoTrack to launch an app on the other side of the client when we activate a devices camera. But the client does not receive the VideoTrack when we enable it with this new version. The client is an android tv devices with Twilio 2.2.1. Is there any retro compatibility ? Or this is the good way to migrate to the latest version ?

#### Code

```swift
    func connectToRoom(_ roomName: String, withToken token: String) {
        let localAudioTrack = TVILocalAudioTrack(options: nil, enabled: false, name: "")
        self.localAudioTrack = localAudioTrack

        guard let frontCamera = TVICameraSource.captureDevice(for: .front) else {
            Log.debug("Front camera not available")
            return
        }
        let cameraSourceOptions = TVICameraSourceOptions { builder in
            builder.enablePreview = true
        }
        camera = TVICameraSource(options: cameraSourceOptions, delegate: self)
        guard let camera = camera else {
            Log.error("Cannot create camera")
            return
        }

        let localVideoTrack = TVILocalVideoTrack(source: camera)
        if localVideoTrack == nil {
            Log.error("Failed to create video track")
        } else {
            self.localVideoTrack = localVideoTrack
        }

        camera.startCapture(with: frontCamera) { captureDevice, _, error in
            if let error = error {
                Log.error("Capture failed with error.\ncode = \((error as NSError).code) error = \(error.localizedDescription)")
                self.camera?.previewView?.removeFromSuperview()
            } else {
                Log.debug("cameraCaptur didStart with captureDevice \(captureDevice.description)")
            }
        }

        let connectOptions = TVIConnectOptions(token: token) { builder in
            if let localAudioTrack = localAudioTrack {
                builder.audioTracks = [localAudioTrack]
            }
            if let localVideoTrack = localVideoTrack {
                builder.videoTracks = [localVideoTrack]
            }
            builder.roomName = roomName
        }

        #if DEBUG
        TwilioVideo.setLogLevel(.debug)
        #endif

        room = TwilioVideo.connect(with: connectOptions, delegate: self)
        statsTimer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(self.statsTimerFired),
                                          userInfo: nil, repeats: true)
    }

    func prepareLocalMedia(cameraSource: AVCaptureDevice) {
        // Create a video track which captures from the front camera.
        if let camera = self.camera {
            if localVideoTrack == nil {
                localVideoTrack = TVILocalVideoTrack(source: camera)
                Log.debug("Video track created.")
            }
            if let preview = camera.previewView {
                Log.debug("Video track created.")
                delegate?.onLocalVideoStreaming(preview)
            }
        }
    }
```

#### Expected Behavior

When we launch the front or back cam the other connected devices would launch an app.

#### Actual Behavior

Nothing.

#### Reproduces How Often

100%

#### Video iOS SDK

2.6.0 via Carthage

#### Xcode

10.14

#### iOS Version

12.1.2

#### iOS Device

All devices

**Comment by ceaglest** [comment]
Hi @Max0u,

Sorry for the late response on my end.

> The client is an android tv devices with Twilio 2.2.1. Is there any retro compatibility ? Or this is the good way to migrate to the latest version ?

Are you referring to Twilio Video [2.2.1](
https://www.twilio.com/docs/video/changelog-twilio-video-android-version-2x#221-september-13-2018) for Android? Any non-deprecated Video Clients are compatible with each other. There are newer versions of Android available like [3.2.2](
https://www.twilio.com/docs/video/changelog-twilio-video-android-3x#322-january-8th-2019), but you are not required to update at this time. Note that upgrading requires that your project use Java 8, which was one reason for the major version bump.

> prepareLocalMedia

This method doesn't appear to be used, and is a duplicate of parts of `connectToRoom`. Calling both `prepareLocalMedia` and `connectToRoom` would cause two video tracks to be created. I don't see any problems with `connectToRoom` per-se, but I can't see how you are handling subscribing to / displaying Tracks on Android or iOS.

Can you please provide us with a Room SID and logs so we can debug the problem further? This can be accessed via the [Room.sid property](https://twilio.github.io/twilio-video-ios/docs/latest/Classes/TVIRoom.html#//api/name/sid).

Regards,
Chris

**Comment by ceaglest** [comment]
Let us know if you need more help.
